### PR TITLE
Update `metrics` and `component.data` packages

### PR DESCRIPTION
**Pull request by tiyash-basu-frequenz**

### CODE REVIEW

**Comment by tiyash-basu-frequenz** [comment]
@thomas-nicolai-frequenz Ready for a look

File: proto/frequenz/api/common/v1/metrics.proto [hunk 1]
```diff
@@ -147,7 +66,7 @@ message AggregatedMetricSample {
 // `AggregatedMetricSample`, but not both simultaneously. Setting one will
 // nullify the other.
 message MetricSampleVariant {
-  oneof metric_return_type {
+  oneof metric_sample_type {
```
Comment (tiyash-basu-frequenz) [inline proto/frequenz/api/common/v1/metrics.proto:69]: `metric_return_type` sounded odd. How does `metric_sample_type` sound?

Comment (thomas-nicolai-frequenz) [reply proto/frequenz/api/common/v1/metrics.proto:69]: lol I never fixed that one. much better! it was just a placeholder as I was missing a better idea at the time!

**Review by thomas-nicolai-frequenz — DISMISSED** [review]

**Comment by tiyash-basu-frequenz** [comment]
@thomas-nicolai-frequenz as you mentioned IRL, I updated the example JSON for `ComponentData`.

**Review by stefan-brus-frequenz — APPROVED** [review]

**Pull request merged**